### DISCUSSION TITLE
Cannot connect to wikijs with ldap account

### DISCUSSION THREAD

**Discussion by TheLaskY** [requarks/wiki#3620] [Error / Bug Report]
### Actual behavior
Created internal wiki and need to connect with ldap (will not create 80+ accounts for users), configured, ldap contacted but cannot login
![image](https://user-images.githubusercontent.com/38154657/95065790-ce27ab00-0701-11eb-8116-f06a102517ea.png)
![login](https://user-images.githubusercontent.com/38154657/95066045-28c10700-0702-11eb-874c-b4332d9118ed.jpeg)

### Expected behavior
Can login to wiki js with ldap account

### Steps to reproduce the behavior
![modified](https://user-images.githubusercontent.com/38154657/95065628-9ae51c00-0701-11eb-8fd8-003332432075.jpeg)
![image](https://user-images.githubusercontent.com/38154657/95065687-b05a4600-0701-11eb-9359-e5226755c1f5.png)

**Comment by flagal** [comment]
In **Admin Bind DN** field, type your email and in **Admin Bind Credential** type the password. **Search Filter** field I put in _(mail={{username}})_. In my case it worked.

**Comment by TheLaskY** [comment]
Changed to those settings and still no connection to the ldap
Do the machine needs to be in the realm?
do i need to create a **proper** id for ldap auth ?
like simple login, password never expire ?

In the logs even if i change the conf wikijs tells me `2020-10-06T06:28:30.203Z [MASTER] info: Authentication Strategy LDAP / Active Directory: [ OK ]`

**Comment by flagal** [comment]
Well, I believe the machine needs to at least access the domain controller. For test purpose, your email from LDAP is sufficient. At my company is works that manner. I don't created a dedicated user for wiki yet. 
This output log don't say nothing error whem you change any config. only when you try to log.

**Comment by TheLaskY** [comment]
Exactly,
I tried to make it join the domain (debian 10 distrib) and even then can't log into wiki js.
Does it means that the problem is not wiki js and in our infrastructure?
Because yea each time i change some configuration, logs are OK.
For the record i was trying to log with a account (not in the wiki user database) and see live log error or not but i can't find a tool / location for **zentyal AD**.
I'll try to simulate some real AD on VMs to diagnose this problem

**Comment by cpasanderlues** [comment]
Same problem here.

I created a dedicated user to bind my Active Directory to WikiJS.
This user doesn't have an email and is set up as follow:

- User can't change password
- Password never expires

I installed WikiJS via Docker on a 18.04 Ubuntu VM.
I'm using the latest version of WikiJS (2.5.159)

My WikiJS LDAP / Active Directory configuration is the following:

- Search Filter: (sAMAccountName={{username}})
- Unique ID Field Mapping: sAMAccountName
- Email Field mapping: mail


I have multiple services using LDAP authentication and I really don't know what's happening here.
Logs telling me:

- info: Authentication Strategy Local Database: [OK]
- info: Authentication Strategy LDAP / Active Directory: [OK]

**Comment by TheLaskY** [comment]
BIG UPDATE
I re-created an environnement Windows Server 2019 and debian 10 (non domain added) with wiki js running into node
Copied/Pasted all parameters from current conf and it worked, but it seems like zentyal is blocking some packets concerning ldap because when i voluntarily used wrong username/pass or mail/pass log says: `LDAP LOGIN ERROR (c2)` which is right because login is wrong!

Note that even with fresh Windows Server, when querying ldap with ldapsearch, mail or email doesn't exist and found **userPrincipalName**
![image](https://user-images.githubusercontent.com/38154657/95307248-af075580-0888-11eb-9b97-dc984207ac03.png)
![image](https://user-images.githubusercontent.com/38154657/95307384-d827e600-0888-11eb-96af-c08472a2dea1.png)

**Comment by juliendms** [comment]
We had issues with our LDAP when our users did not have any mail address associated. It seemed at that time that the mail was required so that LDAP users can self-register and connect to WikiJS. I do not know if it is still the case, but it's worth checking.
All our users (including admin) have now a mail address associated in the LDAP and it's working fine.

**Comment by unicorn667** [comment]
Anybody with a working Active Directory integration? I have tried all variants in this topic but nothing works fine :-(

**Comment by unicorn667** [comment]
Ok Update with this seach filter will work the login

(&(objectcategory=person)(objectclass=user)(|(sAMAccountName={{username}})(mail={{username}})))


found here

https://github.com/Requarks/wiki/issues/1921
